rasa/core/policies/intent_ted_policy.py

```diff
@@ -536,6 +566,115 @@ def _check_unlikely_intent(
+            2. Similarity scores when `label_id` is the wrong label.
+
+        Args:
+            output_scores: Model's predictions for each data point.
+            label_ids: Numerical IDs of labels for each data point.
+
+        Returns:
+            Both buckets of similarity scores grouped by each unique label id.
+        """
+        label_id_scores: Dict[int, List[List[float]]] = defaultdict(list)
+        unique_label_ids = np.unique(label_ids).tolist()
+        if LABEL_PAD_ID in unique_label_ids:
+            unique_label_ids.remove(LABEL_PAD_ID)
+
+        for label_id in unique_label_ids:
+            label_id_scores[label_id] = [[], []]
+
+        for index, all_pos_labels in enumerate(label_ids):
+
+            for candidate_label_id in unique_label_ids:
+                if candidate_label_id in all_pos_labels:
+                    label_id_scores[candidate_label_id][0].append(
+                        output_scores["similarities"][index, 0, candidate_label_id]
+                    )
+                else:
+                    label_id_scores[candidate_label_id][1].append(
+                        output_scores["similarities"][index, 0, candidate_label_id]
+                    )
+
+        # Get only unique scores so that duplicate
+        # trackers created because of permutations are pruned out.
+        unique_label_id_scores = {
+            label_id: (list(set(scores[0])), list(set(scores[1])))
+            for label_id, scores in label_id_scores.items()
+        }
+        return unique_label_id_scores
+
+    @staticmethod
+    def _compute_label_quantiles(
+        label_id_scores: Dict[int, Tuple[List[float], List[float]]]
+    ) -> Dict[int, List[float]]:
+        """Computes multiple quantiles for each label id.
+
+        The quantiles are computed over the negative scores
+        collected for each label id. However, no quantile score
+        can be greater than the minimum positive score collected
+        for the corresponding label id.
+
+        Args:
+            label_id_scores: Scores collected for each label id
+                over positive and negative trackers.
+
+        Returns:
+            Computed quantiles for each label id.
+        """
+        label_quantiles = {}
+
+        for label_id, (positive_scores, negative_scores) in label_id_scores.items():
+            label_quantiles[label_id] = [
+                min(
+                    min(positive_scores),
+                    np.quantile(
+                        negative_scores,
+                        1 - tolerance_value / 100.0,
+                        interpolation="lower",
+                    ),
+                )
+                if negative_scores
+                else min(positive_scores)
+                for tolerance_value in range(0, 100, 5)
+            ]
+
+        return label_quantiles
+
+    @staticmethod
+    def _pick_thresholds(
+        label_quantiles: Dict[int, List[float]], tolerance: float
```

Comment:
I know we check the config to confirm tolerance is in [0, 1] but maybe we should raise an exception here if it is not, in case any one every takes intentTED for a spin outside the standard codebase?

Reply:
Even then, they would be expected to initialize `IntentTEDPolicy` with its constructor or the `load` method (since they are the public methods) which means the check will be applied and will trigger an exception if tolerance is not in accepted ranges.